packages/kit/src/utils/routing.js

```diff
@@ -100,6 +102,23 @@ function affects_path(segment) {
 	return !/^\([^)]+\)$/.test(segment);
 }
 
+/**
+ * Throws error if segment contains invalid characters
+ * @param {string} segment
+ */
+function assert_segment(segment) {
```

Comment:
I'm not exactly sure about this. It's unclear to me what exactly it's disallowing and at the very least would need some comment explaining what is considered to be an invalid character. It's also difficult for users to understand which characters are invalid from the error message. I think we could write this whole thing in a clearer way